src/Functions/keyvaluepair/src/impl/state/StateHandler.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <optional>
+#include <string_view>
+
+namespace DB
+{
+
+struct StateHandler
```

Comment:
What is the purpose of `StateHandler`? From the first look, it has only settings (`escape_character`, `enclosing_character`) and `get()` method without reading anything, and it's not clear what the responsibility is. Should children introduce methods that would affect the state?

Reply:
Hmm... I guess it's easier to explain starting with the derived classes (ValueStateHandler & KeyStateHandler). As you have seen previously, there are a couple of value & key related states (WAITING_KEY, READING_KEY, WAITING_VALUE, READING_VALUE and more...). These classes handle those states. StateHandler, the base class, is just a class that contains the common parts of both. They are: `escape_character`, `enclosing_character`, `get() interface` and `createElement`.

Do you have a suggestion here?

Reply:
Ok, let's just add some comments about `get` method and probably give a bit more specific name what we get (not state, but some data).

Reply:
That does make sense, looking from another perspective `get` gives the idea of `get state`. I'll rename it to `getElement`, but can go for a different name if you have a better suggestion.